scripts/poms/sbndpoms_caf_concatenation.sh

```diff
@@ -0,0 +1,290 @@
+      then
+        WORKDIR="$2"
+        shift
+      else
+        echo "$0 ERROR: workdir requires a non-empty option argument."
+        exit 1
+      fi
+      ;;
+    --outdir)       # Takes an option argument; ensure it has been specified.
+      if [ "$2" ]
+      then
+        OUTDIR="$2"
+        shift
+      else
+        echo "$0 ERROR: outdir requires a non-empty option argument."
+        exit 1
+      fi
+      ;;
+    --size)       # Takes an option argument; ensure it has been specified.
+      if [ "$2" ]
+      then
+        MAXSIZE="$2"
+        shift
+      else
+        echo "$0 ERROR: size requires a non-empty option argument."
+        exit 1
+      fi
+      ;;
+    --flatten)       # Takes an option argument; ensure it has been specified.
+      FLATTEN=true
+      shift
+      ;;
+    *) break
+  esac
+  shift
+done
+
+if  [ -z "$DEFNAME" ]
+then
+  echo "$0 ERROR: defname is mandatory"
+  exit 2
+fi
+
+if  [ -z "$WORKDIR" ]
+then
+  echo "$0 ERROR: workdir is mandatory"
+  exit 2
+fi
+
+if  [ -z "$OUTDIR" ]
+then
+  echo "$0 ERROR: outdir is mandatory"
+  exit 2
+fi
+
+if  [ -z "$MAXSIZE" ]
+then
+  echo "$0 ERROR: size is mandatory"
+  exit 2
+fi
+
+echo "Running with settings:"
+echo "  Definition: $DEFNAME"
+echo "  Workdir: $WORKDIR"
+echo "  Outdir: $OUTDIR"
+echo "  Max Size: $(numfmt --to=si $MAXSIZE)"
+if [ ! -z "$FLATTEN" ]
+then
+  echo "  Making Flat CAFS"
+fi
+
+prepare()
+{
+  # Assume that the metadata is roughly the same in all files, so take the first
+  FILENAME=$(samweb -e sbnd list-definition-files $DEFNAME | head -n 1)
+  samweb -e sbnd get-metadata $FILENAME --json > base.json
+
+  # Lets change "caf" to "concat_caf" for the stage and defname
+  MDPRODUCTIONDEFNAME=$(echo "$DEFNAME" | sed "s/caf/concat_caf/g")
+
+  # Now we want to set some global vars to define the ouput path
+  MDFILETYPE=$(cat base.json | grep "file_type" | cut -d\" -f4)
+  MDPRODUCTIONTYPE=$(cat base.json | grep "production\.type" | cut -d\" -f4)
+  MDPRODUCTIONNAME=$(cat base.json | grep "production\.name" | cut -d\" -f4)
+  MDSBNDPROJECTNAME=$(cat base.json | grep "sbnd_project\.name" | cut -d\" -f4)
+  MDSBNDPROJECTVERSION=$(cat base.json | grep "sbnd_project\.version" | cut -d\" -f4)
+  MDSBNDPROJECTSTAGE=$(cat base.json | grep "sbnd_project\.stage" | cut -d\" -f4)
+
+  OUTDIR="$OUTDIR/$MDFILETYPE/$MDPRODUCTIONTYPE/$MDPRODUCTIONNAME/$MDSBNDPROJECTNAME/$MDSBNDPROJECTVERSION/$MDSBNDPROJECTSTAGE"
+  WORKDIR="$WORKDIR/$MDFILETYPE/$MDPRODUCTIONTYPE/$MDPRODUCTIONNAME/$MDSBNDPROJECTNAME/$MDSBNDPROJECTVERSION/$MDSBNDPROJECTSTAGE"
+
+  mkdir -p $OUTDIR
+  mkdir -p $WORKDIR
+
+  if [ -z "$(ls -A $WORKDIR)" ]
+  then
+    echo "WORKDIR: $WORKDIR"
+  else
+    echo "Not Empty: $WORKDIR"
+    exit 3
+  fi
+
+  if [ -z "$(ls -A $OUTDIR)" ]
+  then
+    echo "OUTDIR: $OUTDIR"
+  else
+    echo "Not Empty: $OUTDIR"
+    exit 3
+  fi
+
+  if [ 0 -ne $(samweb -e sbnd list-definitions | grep "$MDPRODUCTIONDEFNAME" | wc -l) ]
```

Comment:
`grep` already returns a success/failure code. I think you can do `if samweb -e sbnd list-definitions | grep -q $MDPRODUCTIONDEFNAME`